The "result.rewrite-rule.xsl" clean preprocessing extension does not work anymore

I'm using DITA OT 3.6. I publish a small DITA Map to HTML5, I set the "result.rewrite-rule.xsl" parameter to an XSLT file path. The XSLT contains:

    <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
    version="2.0">
    
    <xsl:template match="node() | @*">
        <xsl:copy>
            <xsl:apply-templates select="node() | @*"/>
        </xsl:copy>
    </xsl:template>
    
    <xsl:template match="file[@format = 'image']/@result">
        <xsl:attribute name="{local-name()}" select="concat('images/', .)"/>
    </xsl:template>
    
     </xsl:stylesheet>

The publishing fails:

    clean-preprocess:
    [clean-preprocess] Error 
    [clean-preprocess]   I/O error reported by XML parser processing javax.xml.transform.sax.SAXSource@6ed78a5e:
    [clean-preprocess]   /Users/user/Desktop/Oxygen XML Editor/Oxygen XML
    [clean-preprocess]   Editor.app/javax.xml.transform.sax.SAXSource@6ed78a5e (No such file or directory)
    
    BUILD FAILED
    /Users/user/Desktop/Oxygen XML Editor/frameworks/dita/DITA-OT3.x/plugins/org.dita.base/build.xml:29: The following error occurred while executing this line:
    /Users/user/Desktop/Oxygen XML Editor/frameworks/dita/DITA-OT3.x/plugins/org.dita.base/build_preprocess.xml:347: java.lang.RuntimeException: Failed to compile XSLT: I/O error reported by XML parser processing javax.xml.transform.sax.SAXSource@6ed78a5e: /Users/user/Desktop/Oxygen XML Editor/Oxygen XML Editor.app/javax.xml.transform.sax.SAXSource@6ed78a5e (No such file or directory)
	at org.dita.dost.module.CleanPreprocessModule.lambda$init$1(CleanPreprocessModule.java:88)
	at java.base/java.util.Optional.map(Optional.java:258)
	at org.dita.dost.module.CleanPreprocessModule.init(CleanPreprocessModule.java:78)
	at org.dita.dost.module.CleanPreprocessModule.execute(CleanPreprocessModule.java:110)
	at org.dita.dost.module.AbstractPipelineModuleImpl.execute(AbstractPipelineModuleImpl.java:49)
	at org.dita.dost.ant.ExtensibleAntInvoker.execute(ExtensibleAntInvoker.java:190)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:36)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:437)
	at org.apache.tools.ant.taskdefs.CallTarget.execute(CallTarget.java:106)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1374)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
	at org.apache.tools.ant.Main.runBuild(Main.java:827)
	at org.apache.tools.ant.Main.startAnt(Main.java:223)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:284)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:101)
    Caused by: net.sf.saxon.s9api.SaxonApiException: I/O error reported by XML parser processing javax.xml.transform.sax.SAXSource@6ed78a5e: /Users/user/Desktop/Oxygen XML Editor/Oxygen XML Editor.app/javax.xml.transform.sax.SAXSource@6ed78a5e (No such file or directory)
	at net.sf.saxon.s9api.XsltCompiler.compile(XsltCompiler.java:793)
	at org.dita.dost.module.CleanPreprocessModule.lambda$init$1(CleanPreprocessModule.java:83)
	... 34 more
    Caused by: net.sf.saxon.trans.XPathException: I/O error reported by XML parser processing javax.xml.transform.sax.SAXSource@6ed78a5e: /Users/user/Desktop/Oxygen XML Editor/Oxygen XML Editor.app/javax.xml.transform.sax.SAXSource@6ed78a5e (No such file or directory)
	at net.sf.saxon.event.Sender.sendSAXSource(Sender.java:463)
	at net.sf.saxon.event.Sender.send(Sender.java:167)
	at net.sf.saxon.style.StylesheetModule.sendStylesheetSource(StylesheetModule.java:274)
	at net.sf.saxon.style.StylesheetModule.loadStylesheet(StylesheetModule.java:200)
	at net.sf.saxon.style.Compilation.compileSingletonPackage(Compilation.java:107)
	at net.sf.saxon.s9api.XsltCompiler.compile(XsltCompiler.java:788)
	... 35 more
Caused by: java.io.FileNotFoundException: /Users/user/Desktop/Oxygen XML Editor/Oxygen XML Editor.app/javax.xml.transform.sax.SAXSource@6ed78a5e (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:211)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:153)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:108)
	at java.base/sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:86)
	at java.base/sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:189)
	at org.apache.xerces.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:1040)
	at org.apache.xerces.impl.XMLVersionDetector.determineDocVersion(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.ditang.relaxng.defaults.RelaxDefaultsParserConfiguration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at net.sf.saxon.event.Sender.sendSAXSource(Sender.java:435)
	... 40 more

--- a/src/main/java/org/dita/dost/module/CleanPreprocessModule.java
+++ b/src/main/java/org/dita/dost/module/CleanPreprocessModule.java
@@ -81,7 +81,7 @@ public class CleanPreprocessModule extends AbstractPipelineModuleImpl {
                         final Processor processor = xmlUtils.getProcessor();
                         final XsltCompiler xsltCompiler = processor.newXsltCompiler();
                         xsltCompiler.setErrorListener(toErrorListener(logger));
-                        final XsltExecutable xsltExecutable = xsltCompiler.compile(new StreamSource(f.toString()));
+                        final XsltExecutable xsltExecutable = xsltCompiler.compile(f);
                         return xsltExecutable.load();
                     } catch (UncheckedXPathException e) {
                         throw new RuntimeException("Failed to compile XSLT: " + e.getXPathException().getMessageAndLocation(), e);

--- a/src/test/java/org/dita/dost/module/CleanPreprocessModuleTest.java
+++ b/src/test/java/org/dita/dost/module/CleanPreprocessModuleTest.java
@@ -8,17 +8,23 @@
 
 package org.dita.dost.module;
 
+import static java.net.URI.create;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.dita.dost.TestUtils;
+import org.dita.dost.TestUtils.TestLogger;
 import org.dita.dost.store.StreamStore;
 import org.dita.dost.util.Job;
 import org.dita.dost.util.Job.FileInfo.Builder;
 import org.dita.dost.util.XMLUtils;
 import org.junit.Test;
-
-import java.io.File;
-import java.net.URI;
-
-import static java.net.URI.create;
-import static org.junit.Assert.assertEquals;
 
 public class CleanPreprocessModuleTest {
 
@@ -116,4 +122,35 @@ public class CleanPreprocessModuleTest {
         assertEquals(create("file:/foo/bar/"), module.getBaseDir());
     }
 
+    @Test
+    public void reviewRewrite() throws Exception {
+        final File tempDir = new File("").getAbsoluteFile();
+        final Job job = new Job(tempDir, new StreamStore(tempDir, new XMLUtils()));
+        job.setInputDir(URI.create("file:/foo/bar/"));
+        job.add(new Builder()
+                .uri(create("map.ditamap"))
+                .isInput(true)
+                .result(create("file:/foo/bar/map.ditamap"))
+                .build());
+        job.add(new Builder()
+                .uri(create("topics/topic.dita"))
+                .result(create("file:/foo/bar/topics/topic.dita"))
+                .build());
+        module.setJob(job);
+        module.setXmlUtils(new XMLUtils());
+        TestLogger logger = new TestUtils.TestLogger(false);
+        module.setLogger(logger);
+        Map<String, String> input = new HashMap<>();
+        input.put("use-result-filename", "true");
+        input.put("result.rewrite-rule.xsl", "abc.xsl");
+        String errorMessage = null;
+        try {
+        	module.execute(input);
+        } catch(Exception ex) {
+        	ex.printStackTrace();
+        	errorMessage = ex.getMessage();
+        }
+        assertTrue("Should have properly tried to locate the abc.xsl which does not exist.", 
+        		errorMessage.contains("Failed to compile XSLT") && errorMessage.contains("abc.xsl"));
+    }
 }

--- a/src/test/java/org/dita/dost/module/CleanPreprocessModuleTest.java
+++ b/src/test/java/org/dita/dost/module/CleanPreprocessModuleTest.java
@@ -8,9 +8,16 @@
 
 package org.dita.dost.module;
 
-import static java.net.URI.create;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import org.dita.dost.TestUtils;
+import org.dita.dost.TestUtils.TestLogger;
+import org.dita.dost.store.StreamStore;
+import org.dita.dost.util.Job;
+import org.dita.dost.util.Job.FileInfo.Builder;
+import org.dita.dost.util.XMLUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
 
 import java.io.File;
 import java.io.IOException;
@@ -18,17 +25,26 @@ import java.net.URI;
 import java.util.HashMap;
 import java.util.Map;
 
-import org.dita.dost.TestUtils;
-import org.dita.dost.TestUtils.TestLogger;
-import org.dita.dost.store.StreamStore;
-import org.dita.dost.util.Job;
-import org.dita.dost.util.Job.FileInfo.Builder;
-import org.dita.dost.util.XMLUtils;
-import org.junit.Test;
+import static java.net.URI.create;
+import static org.junit.Assert.assertEquals;
 
 public class CleanPreprocessModuleTest {
 
-    private final CleanPreprocessModule module = new CleanPreprocessModule();
+    private CleanPreprocessModule module;
+    private XMLUtils xmlUtils;
+    private Job job;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Before
+    public void setUp() throws IOException {
+        module = new CleanPreprocessModule();
+        xmlUtils = new XMLUtils();
+        final File tempDir = temporaryFolder.newFolder();
+        job = new Job(tempDir, new StreamStore(tempDir, xmlUtils));
+        module.setJob(job);
+    }
 
     @Test
     public void getCommonBase() {
@@ -41,8 +57,6 @@ public class CleanPreprocessModuleTest {
 
     @Test
     public void getBaseDir() throws Exception {
-        final File tempDir = new File("").getAbsoluteFile();
-        final Job job = new Job(tempDir, new StreamStore(tempDir, new XMLUtils()));
         job.setInputDir(URI.create("file:/foo/bar/"));
         job.add(new Builder()
                 .uri(create("map.ditamap"))
@@ -64,14 +78,12 @@ public class CleanPreprocessModuleTest {
                 .uri(create("common/topic.dita"))
                 .result(create("file:/foo/bar/common/topic.dita"))
                 .build());
-        module.setJob(job);
+
         assertEquals(create("file:/foo/bar/"), module.getBaseDir());
     }
 
     @Test
     public void getBaseDirExternal() throws Exception {
-        final File tempDir = new File("").getAbsoluteFile();
-        final Job job = new Job(tempDir, new StreamStore(tempDir, new XMLUtils()));
         job.setInputDir(URI.create("file:/foo/bar/"));
         job.add(new Builder()
                 .uri(create("map.ditamap"))
@@ -82,14 +94,12 @@ public class CleanPreprocessModuleTest {
                 .uri(create("topics/topic.dita"))
                 .result(create("https://example.com/topics/bar/topics/topic.dita"))
                 .build());
-        module.setJob(job);
+
         assertEquals(create("file:/foo/bar/"), module.getBaseDir());
     }
 
     @Test
     public void getBaseDirSubdir() throws Exception {
-        final File tempDir = new File("").getAbsoluteFile();
-        final Job job = new Job(tempDir, new StreamStore(tempDir, new XMLUtils()));
         job.setInputDir(URI.create("file:/foo/bar/maps/"));
         job.add(new Builder()
                 .uri(create("maps/map.ditamap"))
@@ -100,14 +110,12 @@ public class CleanPreprocessModuleTest {
                 .uri(create("topics/topic.dita"))
                 .result(create("file:/foo/bar/topics/topic.dita"))
                 .build());
-        module.setJob(job);
+
         assertEquals(create("file:/foo/bar/"), module.getBaseDir());
     }
 
     @Test
     public void getBaseDirSupdir() throws Exception {
-        final File tempDir = new File("").getAbsoluteFile();
-        final Job job = new Job(tempDir, new StreamStore(tempDir, new XMLUtils()));
         job.setInputDir(URI.create("file:/foo/bar/maps/"));
         job.add(new Builder()
                 .uri(create("maps/map.ditamap"))
@@ -118,39 +126,19 @@ public class CleanPreprocessModuleTest {
                 .uri(create("topics/topic.dita"))
                 .result(create("file:/foo/bar/topic.dita"))
                 .build());
-        module.setJob(job);
+
         assertEquals(create("file:/foo/bar/"), module.getBaseDir());
     }
 
-    @Test
-    public void reviewRewrite() throws Exception {
-        final File tempDir = new File("").getAbsoluteFile();
-        final Job job = new Job(tempDir, new StreamStore(tempDir, new XMLUtils()));
-        job.setInputDir(URI.create("file:/foo/bar/"));
-        job.add(new Builder()
-                .uri(create("map.ditamap"))
-                .isInput(true)
-                .result(create("file:/foo/bar/map.ditamap"))
-                .build());
-        job.add(new Builder()
-                .uri(create("topics/topic.dita"))
-                .result(create("file:/foo/bar/topics/topic.dita"))
-                .build());
+    @Test(expected = RuntimeException.class)
+    public void RewriteRule_WhenStylesheetNotFound_ShouldThrowException() throws Exception {
         module.setJob(job);
-        module.setXmlUtils(new XMLUtils());
-        TestLogger logger = new TestUtils.TestLogger(false);
+        module.setXmlUtils(xmlUtils);
+        final TestLogger logger = new TestUtils.TestLogger(false);
         module.setLogger(logger);
-        Map<String, String> input = new HashMap<>();
-        input.put("use-result-filename", "true");
+        final Map<String, String> input = new HashMap<>();
         input.put("result.rewrite-rule.xsl", "abc.xsl");
-        String errorMessage = null;
-        try {
-        	module.execute(input);
-        } catch(Exception ex) {
-        	ex.printStackTrace();
-        	errorMessage = ex.getMessage();
-        }
-        assertTrue("Should have properly tried to locate the abc.xsl which does not exist.", 
-        		errorMessage.contains("Failed to compile XSLT") && errorMessage.contains("abc.xsl"));
+
+        module.execute(input);
     }
 }

